src/main/java/com/hazelcast/remotecontroller/CloudManager.java

```diff
@@ -0,0 +1,436 @@
+    }
+
+    // For a deleted cluster we cannot use the waitForStateOfCluster() method, its logic is different. That is why another method is added for delete cluster process
+    private void waitForDeletedCluster(String clusterId, long timeoutInMillisecond) throws CloudException, InterruptedException {
+        long startTime = System.currentTimeMillis();
+        while (true) {
+            try {
+                // Don't setup tls, we just want to check if cluster is deleted
+                getCloudCluster(clusterId, false);
+            } catch (CloudException e) {
+                // Cluster is deleted
+                return;
+            }
+            if ((System.currentTimeMillis() - startTime) + TimeUnit.SECONDS.toMillis(RETRY_TIME_IN_SECOND)  > timeoutInMillisecond) {
+                break;
+            }
+            TimeUnit.SECONDS.sleep(RETRY_TIME_IN_SECOND);
+        }
+        throw new CloudException(String.format("The cluster with id %s is not deleted in %d millisecond", clusterId, timeoutInMillisecond));
+    }
+
+    private String getBearerToken(String apiKey, String apiSecret) throws CloudException {
+        try {
+            String query = String.format("{ \"apiKey\": \"%s\", \"apiSecret\": \"%s\" }", apiKey, apiSecret);
+            LOG.info(String.format("Sending query %s for getting bearer token", query));
+            RequestBody body = RequestBody.create(JSON, query);
+            Request request = new Request.Builder()
+                    .url(baseUrl + "/customers/api/login")
+                    .post(body)
+                    .header("Content-Type", "application/json")
+                    .build();
+            call = CLIENT.newCall(request);
+            Response response = call.execute();
+            ResponseBody responseBody = response.body();
+            if (responseBody == null) {
+                throw new CloudException(String.format("Response body is null while getting bearer token: %s", response));
+            }
+            String responseString = responseBody.string();
+            throwIfResponseFailed(response, responseString);
+            LOG.info(maskValueOfToken(responseString));
+            return mapper.readTree(responseString).get("token").asText();
+        } catch (Exception e) {
+            throw new CloudException("Get bearer token is failed. " + getErrorString(e));
+        }
+    }
+
+    // It replaces the value of token with *** in a response body. Such as bearer token, discovery token.
+    private String maskValueOfToken(String json)
+    {
+        return json.replaceAll("\"token\"\\s*:\\s*\"(.*)\"", "\"token\":\"******\"");
+    }
+
+    // It creates a folder with name clusterId under /home/user/
+    // Then it downloads the certificates for the cluster and put them in the created folder
+    private String downloadCertificatesAndGetPath(String clusterId) throws CloudException {
+        try
+        {
+            String userHome = System.getProperty("user.home");
```

Comment:
How is it work in Windows?

Reply:
It will return user home directory. https://stackoverflow.com/questions/42184176/what-is-system-getpropertyuser-home-output-in-windows

I did not change this logic actually. It was preexistent. What do you mean by your second question?

Reply:
Okay, it seems okay for windows. By second question, I mean it can download the files into same directory with jar file.

Reply:
Yeah we could do that, but Ali Sengul did not choose to do that.